lisa/sut_orchestrator/azure/features.py

```diff
@@ -2009,8 +2016,11 @@ def create_setting(
 
     @classmethod
     def _enable_hibernation(cls, *args: Any, **kwargs: Any) -> None:
-        parameters: Any = kwargs.get("arm_parameters")
-        if parameters.use_availability_sets:
+        parameters = cast(AzureArmParameter, kwargs.get("arm_parameters"))
+        if (
+            parameters.availability_options.availability_type
```

Comment:
@squirrelsc this logic is running in Hibernation.on_before_deployment. Are there any guarantees about what order the Feature.on_before_deployment will be run? If Hibernation runs before Availability, the arm parameters might not be accurate for this check.

Reply:
You can change it to get availability set information from `environment`, instead of the `arm_parameters`.

Reply:
With the current implementation, this check in Hibernation is dependent on the code run in Availability.on_befroe_deployment. Checking the environment would also not be accurate.

Availability was implemented this way because it has a dependency on data_disk_type / Ultra Disk usage.

Reply:
This may raise exception during deployment. If you find it, it can be skipped in hooks.py.

Reply:
Please see my comment below
> Is this an acceptable solution? It will guarantee the order of on_before_deployment will match the order of the list returned by AzurePlatform.supported_features() Do you foresee this change causing any problems? Is it a bad idea to rely on a guaranteed order of execution?
> 
> This felt like a better solution to me than trying to catch specific deployment errors in hooks.py